src/read.c

```diff
@@ -3962,6 +4108,71 @@ avifResult avifDecoderReset(avifDecoder * decoder)
             }
         }
 
+#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
+        avifDecoderItem * gainMapItem = NULL;
+        avifDecoderItem * toneMappedImageItem;
+        uint32_t gainMapItemID;
+        AVIF_CHECKRES(avifDecoderDataFindToneMappedImageItem(data, colorItem, &toneMappedImageItem, &gainMapItemID));
+        avifCodecType gainMapCodecType = AVIF_CODEC_TYPE_UNKNOWN;
+        if (toneMappedImageItem) {
+            avifROData tmapData;
+            AVIF_CHECKRES(avifDecoderItemRead(toneMappedImageItem, decoder->io, &tmapData, 0, 0, data->diag));
+            AVIF_CHECKERR(avifParseToneMappedImageBox(&decoder->image->gainMap.metadata, tmapData.data, tmapData.size, data->diag),
+                          AVIF_RESULT_INVALID_TONE_MAPPED_IMAGE);
+
+            gainMapItem = avifMetaFindItem(data->meta, gainMapItemID);
+            if (!gainMapItem) {
+                avifDiagnosticsPrintf(data->diag, "Box[tmap] gain map item ID %d not found", gainMapItemID);
+                return AVIF_RESULT_INVALID_TONE_MAPPED_IMAGE;
+            }
+
+            if (!memcmp(gainMapItem->type, "grid", 4)) {
+                avifROData gainMapGridData;
+                AVIF_CHECKRES(avifDecoderItemRead(gainMapItem, decoder->io, &gainMapGridData, 0, 0, data->diag));
+                AVIF_CHECKERR(avifParseImageGridBox(&data->gainMap.grid,
+                                                    gainMapGridData.data,
+                                                    gainMapGridData.size,
+                                                    decoder->imageSizeLimit,
+                                                    decoder->imageDimensionLimit,
+                                                    data->diag),
+                              AVIF_RESULT_INVALID_IMAGE_GRID);
+                gainMapCodecType = avifDecoderItemGetGridCodecType(gainMapItem);
+                if (gainMapCodecType == AVIF_CODEC_TYPE_UNKNOWN) {
+                    return AVIF_RESULT_INVALID_IMAGE_GRID;
+                }
+            } else {
+                gainMapCodecType = avifGetCodecType(gainMapItem->type);
+                assert(gainMapCodecType != AVIF_CODEC_TYPE_UNKNOWN);
+            }
+
+            decoder->image->gainMap.image = avifImageCreateEmpty();
+
+            // Look for a colr nclx box. Other colr box types (e.g. ICC) are not supported.
```

Comment:
AVIF_RESULT_NOT_IMPLEMENTED suggests that this is a valid use case and that it's not currently implemented, but might be in the future, right?

I don't think an icc profile makes sense for the gain map, so AVIF_RESULT_NOT_IMPLEMENTED does not seem right to me.
The draft spec says `The gain map input image shall be associated with a colr item property of type nclx.` Not sure if that's supposed to imply if other types are forbidden or not (in addition to nclx).

Should I ignore other boxes (like I'm doing now, since I don't know what to do with them) or return some parsing error (AVIF_RESULT_BMFF_PARSE_FAILED ?) or return AVIF_RESULT_NOT_IMPLEMENTED assuming that maybe it's a new type of box that I don't support?